lib/shopify_app/session/session_storage.rb

```diff
@@ -3,7 +3,8 @@ module SessionStorage
     extend ActiveSupport::Concern
 
     included do
-      validates :shopify_domain, presence: true, uniqueness: { case_sensitive: false }
+      validates :shopify_domain, presence: true, uniqueness: { case_sensitive: false },
+        unless: Proc.new {|_| ShopifyApp.configuration.per_user_tokens }
```

Comment:
Why do we not need a domain for per user session? How will we identify the domain that the API requests are sent to?

Reply:
In per-user session - many users will have the same domain. In per-shop session, the domain is the unique identifier.

Reply:
from @tanema Does the `unless` apply just to the uniqueness, or to the whole validation statement? Perhaps needs to be split into multiple validates.

Reply:
This was split into multiple validations.